core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/gasprice/gasprice.go

```diff
@@ -26,22 +26,21 @@ type UpkeepOffchainConfig struct {
 
 // CheckGasPrice retrieves the current gas price and compare against the max gas price configured in upkeep's offchain config
 // any errors in offchain config decoding will result in max gas price check disabled
-func CheckGasPrice(ctx context.Context, upkeepId *big.Int, oc []byte, ge gas.EvmFeeEstimator, lggr logger.Logger) encoding.UpkeepFailureReason {
-	if len(oc) == 0 {
+func CheckGasPrice(ctx context.Context, upkeepId *big.Int, offchainConfigBytes []byte, ge gas.EvmFeeEstimator, lggr logger.Logger) encoding.UpkeepFailureReason {
+	if len(offchainConfigBytes) == 0 {
 		return encoding.UpkeepFailureReasonNone
 	}
 
 	var offchainConfig UpkeepOffchainConfig
-	if err := cbor.ParseDietCBORToStruct(oc, &offchainConfig); err != nil {
+	if err := cbor.ParseDietCBORToStruct(offchainConfigBytes, &offchainConfig); err != nil {
 		lggr.Errorw("failed to parse upkeep offchain config, gas price check is disabled", "upkeepId", upkeepId.String(), "err", err)
 		return encoding.UpkeepFailureReasonNone
 	}
-	if offchainConfig.MaxGasPrice == nil {
-		lggr.Infow("maxGasPrice is not configured in upkeep offchain config, gas price check is disabled", "upkeepId", upkeepId.String())
+	if offchainConfig.MaxGasPrice == nil || offchainConfig.MaxGasPrice.Int64() <= 0 {
+		lggr.Warnw("maxGasPrice is not configured or incorrectly configured in upkeep offchain config, gas price check is disabled", "upkeepId", upkeepId.String())
```

Comment:
is my understanding correct in that this would get logged for every upkeep that doesn't have a MaxGasPrice set? Do we think that will be a noisy log for the node?

Reply:
upkeeps without offchain config will exit at line 31. currently all upkeeps should have just empty offchain config.

Reply:
but agree that we can update this in case there are more use cases for offchain config